pkg/util/cluster/cluster.go

```diff
@@ -305,20 +350,26 @@ func (c *Cluster) Create(ctx context.Context, vnetResourceGroup, clusterName str
 		if err != nil {
 			return err
 		}
+		// Only do this if CI=True and Public Cloud
+		c.log.Info("peering subnets to CI infra")
+		err = c.peerSubnetsToCI(ctx, vnetResourceGroup, clusterName)
+		if err != nil {
+			return err
+		}
+
 	}
 
 	c.log.Info("done")
 	return nil
 }
 
 func (c *Cluster) generateSubnets() (vnetPrefix string, masterSubnet string, workerSubnet string) {
-	// pick a random /23 in the range [10.0.2.0, 10.128.0.0).  10.0.0.0 is used
-	// by dev-vnet to host CI; 10.128.0.0+ is used for pods.
+	// pick a random /23 in the range [172.16.0.0, 172.31.255.255]
 	var x, y int
+	rand.Seed(time.Now().UnixNano())
 	for x == 0 && y == 0 {
-		x, y = rand.Intn(128), 2*rand.Intn(128)
+		x, y = rand.Intn((16))+3, 2*rand.Intn(128)
```

Comment:
Select only from `10.3.0.0` All address spaces before that are used by RPs dev-vnet , dev-vpn-vnet and rp-vnet